core/types/block.go

```diff
@@ -594,9 +594,9 @@ func CopyHeader(h *Header) *Header {
 	cpy.number = make([]*big.Int, common.HierarchyDepth)
 	cpy.baseFee = make([]*big.Int, common.HierarchyDepth)
 	for i := 0; i < common.HierarchyDepth; i++ {
-		cpy.difficulty[i] = h.Difficulty(i)
-		cpy.number[i] = h.Number(i)
-		cpy.baseFee[i] = h.BaseFee(i)
+		cpy.SetDifficulty(new(big.Int).Set(h.Difficulty(i)), i)
```

Comment:
you don't need to (and shouldn't) create a new `big.Int` here. `SetDifficulty()` already deep copies the int for you

Reply:
You are right. Changed it.